osu.Game.Rulesets.Osu/Difficulty/MathUtil/Mean.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+
+namespace osu.Game.Rulesets.Osu.Difficulty.MathUtil
+{
+    public static class Mean
+    {
+        public static double PowerMean(double x, double y, double i) =>
+            Math.Pow((Math.Pow(x, i) + Math.Pow(y, i)) / 2, 1 / i);
+
+        public static double PowerMean(IEnumerable<double> values, double i)
```

Comment:
I see that wikipedia uses `p` as the power variable instead of `i`, probably worth changing from `i` at least to avoid confusion with an index

Reply:
This should all be xmldoc'd with references to formulae anyway. I'll grab all of this sort of stuff once I do a pass over this.